tests/integrationv2/fixtures.py

```diff
@@ -6,13 +6,43 @@
 
 from global_flags import get_flag
 from processes import ManagedProcess
-from providers import Provider, S2N
+from providers import Provider, S2N, JavaSSL
 
 from common import ProviderOptions
 
 
 @pytest.fixture
-def managed_process():
+def path_configuration(request: pytest.FixtureRequest):
+    """
+    Determine available providers, and their correct paths if available.
+
+    Currently only supports s2nc/s2nd and the Java SSL client.
+    """
+    if not request.config.getoption("--best-effort"):
+        return None
+
+    providers = {}
+
+    # s2n-tls MUST be available, and we expect it to be in
+    # <git_root>/build/bin
+    expected_location = "../../build/bin"
```

Comment:
Does this work with nix because s2nd/s2nc are just added to PATH? Couldn't we just update `sys.path` within pytest itself to always look for s2nd/s2nc in `../../build/bin`? 

It also seems strange to tie the best-effort feature with finding s2nd/s2nc. Like, if someone is trying to run the integration tests outside of nix without best effort, shouldn't s2nd/s2nc be found in the same way? Can we just always look for s2nd/s2nc in `../../build/bin`?

Reply:
Updating sys.path also seems better than passing around "s2nc" and "s2nd" internally and then modifying the command line arguments to resolve their full path before spawning the process.